### ISSUE TITLE
pSMAC with intensify

### ISSUE THREAD

**Issue by mlindauer** [automl/SMAC3#171]
Hi, 

I wonder whether pSMAC works with the current intensify implementation right now.
I would expect that in pSMAC, a challenger can have more runs than the incumbent.
This should raise an error in `Intensifier_compare_configs`
because the code assumes that the incumbent has at least all runs (insts+seed) of the challenger:

```
        inc_runs = run_history.get_runs_for_config(incumbent)
        chall_runs = run_history.get_runs_for_config(challenger)

        # performance on challenger runs
        chal_perf = aggregate_func(
            challenger, run_history, chall_runs)
        inc_perf = aggregate_func(
            incumbent, run_history, chall_runs)
```

Is this an issue?

**Comment by mfeurer** [comment]
Let me try to comment what should happen according to the code:

1. The challenger will have more runs than the incumbent, but there might be some for which the incumbent was run, but not the challenger. Therefore _race_challenger will [schedule new runs for the challenger](https://github.com/automl/SMAC3/blob/master/smac/intensification/intensification.py#L278).
2. The challenger will no be compared to the incumbent where two things can happen:
    1. The challenger is better than the incumbent, but was run on a different set of instances than the incumbent. It will still become the new incumbent and this might lead to overtuning.
    2. The challenger is worse than the incumbent. Nothing happens and SMAC goes on with the next challenger.

To conclude, instead of checking [the number of runs for the challenger and comparing it to the number of runs for the incumbent](https://github.com/automl/SMAC3/blob/master/smac/intensification/intensification.py#L278), we need to check whether the challenger was run on all instance-seed pairs that the incumbent was run on.

Does this answer your question?

**Comment by mlindauer** [comment]
I agree more or less with your solution.
But the issues is more severe since the call of 

`inc_perf = aggregate_func(incumbent, run_history, chall_runs)`

will fail and raise an exception.

**Comment by mfeurer** [comment]
Okay, I see the issue. So the third argument should be the set of instance-seed pairs on which both the incumbent and the challenger were run, right?

And pSMAC will right now only fail for the multi-instance case, right?

**Comment by mlindauer** [comment]
Yes and yes (and for non-deterministic single-instance )

**Comment by mfeurer** [comment]
Fixed by PR #173
